### ISSUE TITLE
F# Method signature

### ISSUE THREAD

**Issue by CeciAc** [dotnet/dotnet-api-docs#2151]
@SLAVONchick commented on [Tue Mar 26 2019](https://github.com/dotnet/dotnet-api-docs.ru-ru/issues/17)

Consider edit method signature in F# view mode `static member TryParse : string *  -> bool` to `static member TryParse: string -> bool * int`. And other signatures so that it return `bool * int`

---
#### Сведения о документе

⚠ *Не вносите правки в этот раздел. Это необходимо для связывания страницы сайта docs.microsoft.com с вопросом на GitHub.*

* ID: 04a9b653-1243-1bad-96f9-975cbfac4216
* Version Independent ID: 0f2d885e-e6e5-ad21-c946-5480b37d2ba1
* Content: [Int32.TryParse Method (System)](https://docs.microsoft.com/ru-ru/dotnet/api/system.int32.tryparse?view=netframework-4.7.2#feedback)
* Content Source: [xml/System/Int32.xml](https://github.com/dotnet/dotnet-api-docs.ru-ru/blob/live/xml/System/Int32.xml)
* Product: **dotnet-api**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

**Comment by CeciAc** [comment]
Hi team, I'm moving this here since the user is making a suggestion on the example
best regards

**Comment by mairaw** [comment]
Thanks @CeciAc. @cartermp can you look at this feedback related to F# signatures?

**Comment by danieljsummers** [comment]
This is correct; the type signature of `x` in `let x = System.Int32.TryParse ""` is `bool * int`. This should be pretty standard through all the `TryParse` methods, as that's how the `out` parameters are represented in F#.

**Comment by cartermp** [comment]
@danieljsummers Not quite. `out` parameters (that is, `byrefs` that are intended to give a result) will implicitly convert to a tuple for the `TryParse` methods, but the signature is not a tuple of `string -> bool * int`; it's `string -> byref<int> -> bool`.

**Comment by danieljsummers** [comment]
Thanks for clarifying.

Is there a way within the current docs structure to document those differences? I literally wrote code before replying on this thread. :) If that's an overall concept within the language, I didn't realize it; I just learned that "this is the way the `TryParse` methods work".

**Comment by carlossanlop** [comment]
@gewarren @joelmartinez @mimisasouvanh seems like an infra issue with how the F# signature is generated. Do we have an existing internal issue tracking this?

**Comment by mimisasouvanh** [comment]
@danieljsummers @carlossanlop Is this still an issue? Could you clarify what the suggested fix is for this? cc @gewarren @BillWagner @CeciAc

**Comment by danieljsummers** [comment]
@mimisasouvanh The docs still show it as `static member TryParse : string * int -> bool`, whereas @cartermp [said](#issuecomment-529980831) that it should be `string -> byref<int> -> bool` - so I'd say that it is still an issue. I don't know if just changing the signature would help, though, and some code would probably show why. Here's how you would call that in C#...

```csharp
if (int.TryParse(inputVariable, out var parsed)) // do something
```
So, the function call takes two parameters (one `out`), and returns bool - which is what the F# signature above means. However, in practice, it's used differently...

```fsharp
match int.TryParse(inputVariable) with
| true, parsed -> // do something
| false, _ -> // do nothing, or something else
```
This is the heart of the confusion for F# developers - how does the second part of the tuple input parameter in C# become the last parameter of the tuple return value (what that construct matches) in F#?

The suggested fix (at least as I see it) would be either clarifying the signature, or adding an F# example that could explain this at a top-of-the-sample comment. ("F# implicitly swaps these parameters around so they work nicely with pattern matching" or something to that effect.) I'd be happy to convert the C# example to F# if you think it would be valuable (though I'm sure Phillip could do that in his sleep), and the process should be very similar for all the other `TryParse*` methods.

@carlossanlop

**Comment by mimisasouvanh** [comment]
I created an [internal work item](https://ceapex.visualstudio.com/Engineering/_workitems/edit/481150) to track this.
